workspaces/orchestrator/plugins/orchestrator-form-widgets/src/FormWidgetsApi.tsx

```diff
@@ -86,22 +97,144 @@ const safeSet: (errors: JsonObject, path: string, value: JsonValue) => void = (
     safeSet(safeObject, steps[1], value);
   }
 };
+
+type ScmApi = OAuthApi & ProfileInfoApi;
+type ScmOpenIdApi = ScmApi & OpenIdConnectApi;
+
 export class FormWidgetsApi implements OrchestratorFormApi {
-  // private readonly configApi: ConfigApi;
-  // private readonly fetchApi: FetchApi;
+  private readonly fetchApi: FetchApi;
+  private readonly identityApi: IdentityApi;
+  private readonly configApi: ConfigApi;
+
+  private readonly scmApis: { [key: string]: ScmApi };
+  private readonly scmOpenIdApis: { [key: string]: ScmOpenIdApi };
+
+  public constructor(options: {
+    fetchApi: FetchApi;
+    identityApi: IdentityApi;
+    configApi: ConfigApi;
+    githubAuthApi: ScmApi;
+    atlassianAuthApi: ScmApi;
+    googleAuthApi: ScmOpenIdApi;
+    microsoftAuthApi: ScmOpenIdApi;
+    gitlabAuthApi: ScmOpenIdApi;
+  }) {
+    this.fetchApi = options.fetchApi;
+    this.identityApi = options.identityApi;
+    this.configApi = options.configApi;
+
+    this.scmApis = {
+      githubAuthApi: options.githubAuthApi,
+      atlassianAuthApi: options.atlassianAuthApi,
+    };
+    this.scmOpenIdApis = {
+      googleAuthApi: options.googleAuthApi,
+      microsoftAuthApi: options.microsoftAuthApi,
+      gitlabAuthApi: options.gitlabAuthApi,
+    };
+  }
+
+  private async templateUnitEvaluatorIdentityApi(key: string) {
+    if (key === 'token') {
+      return (await this.identityApi.getCredentials()).token;
+    }
+    if (key === 'userEntityRef') {
+      return (await this.identityApi.getBackstageIdentity()).userEntityRef;
+    }
+    if (key === 'profileEmail') {
+      return (await this.identityApi.getProfileInfo()).email;
+    }
+    if (key === 'displayName') {
+      return (await this.identityApi.getProfileInfo()).displayName;
```

Comment:
what about moving this up and reuse the returned value? not sure if calling 2 times the same async endpoint is intendend.
maybe you can consider to use `case` to reduce the cognitive load, but not a strong opinion.